djreact/djangoapps/trips/management/commands/generate_trips.py

```diff
@@ -0,0 +1,206 @@
+        activities_names_list = [
+            'Hiking', 'Snow Fight', 'Camping', 'SightSeeing', 'Trekking',
+            'Skiing', 'Paragliding', 'Cliff Diving', 'Beaches'
+        ]
+        activities_objects_list = []
+        for activity_name in random.sample(activities_names_list, random.choice(range(2, 5))):
+            activity, __ = Activity.objects.get_or_create(
+                name=activity_name
+            )
+            activities_objects_list.append(activity)
+        return activities_objects_list
+
+    @staticmethod
+    def get_random_facilities():
+        """
+        get random number of facilities from some pre-defined facilities
+        """
+        facilities_names_list = [
+            'Transport', 'Meals', 'Guide', 'Photography', 'Accommodation',
+            'First Aid Kit', 'Bon Fire', 'Power Bank'
+        ]
+        facilities_objects_list = []
+        for facility_name in random.sample(facilities_names_list, random.choice(range(2, 5))):
+            facility, __ = Facility.objects.get_or_create(
+                name=facility_name
+            )
+            facilities_objects_list.append(facility)
+        return facilities_objects_list
+
+    @staticmethod
+    def get_random_schedules(trip_duration):
+        """
+        get random schedules datetime objects separated by appropriate trip duration.
+        """
+        schedules_list = []
+        for data_range in range(1, random.choice(range(3, 6))):
+            schedule = datetime.now() + timedelta(days=trip_duration + (data_range*7))
+            schedules_list.append(schedule)
+        return schedules_list
+
+    @staticmethod
+    def get_random_itineraries():
+        """
+        generate random itineraries for a trip with a defined format
+        """
+        itineraries_list = []
+        for day_number in range(1, random.choice(range(4, 7))):
+            day = day_number
+            description = "Itinerary for Day: {}".format(day)
+            itineraries_list.append((day, description))
+        return itineraries_list
+
+    @staticmethod
+    def get_random_gear():
+        """
+        get random number of gears for a trip.
+
+        returns a comma-separated string of gears from a pre-defined
+        list of gears
+        """
+        gears_list = [
+            'Mountain Climber', 'Shoes', 'Stick', 'Coat', 'Camp',
+            'Inhaler', 'Lighter'
+        ]
+        selected_gear = random.sample(gears_list, random.choice(range(1,4)))
+        return ','.join(selected_gear)
+
+    def handle(self, *args, **options):
+        """
+        Generating trip based on the input.
+        """
+        batch_size = options['batch_size']
+        # user required to create a trip
+        user = User.objects.get(username='admin')
+        if not user:
+            raise CommandError(
+                "Username: admin doesn't exist. Run 'make import' to populate base data"
+            )
+
+        for count in range(0, batch_size):
```

Comment:
Just an fyi -- If you would use `xrange` instead of `range` it would return an iterator rather than a static list.